forge-ai/src/main/java/forge/ai/AiController.java

```diff
@@ -562,6 +566,14 @@ private Card chooseBestLandToPlay(CardCollection landList) {
                     if (reSA == null || !ApiType.Tap.equals(reSA.getApi())) {
                         continue;
                     }
+                    // check unlesscost parameters. At this point, it should be a tapapi ability
+                    if (reSA.hasParam("UnlessCost")) {
+                        String unlessCost = reSA.getParam("UnlessCost").trim();
+                        Cost cost = calculateUnlessCost(reSA, unlessCost, true);
+                        if (SpellApiToAi.Converter.get(ApiType.Tap).willPayUnlessCost(reSA, player, cost, false, new PlayerCollection(player))) {
```

Comment:
`SpellApiToAi.Converter.get(reSA)` 
no need for extra Type import

Also, i don't like how `calculateUnlessCost` gets extra imported

Reply:
Thanks for catching and handling this!